windows.js

```diff
@@ -199,24 +199,17 @@ async function setupMSWin() {
  *   adds a convenience VCVARS environment variable
  *   this assumes a single Visual Studio version being available in the Windows images */
 export function addVCVARSEnv() {
-  let vcVars = ''
-  switch (virtualEnv) {
-    case 'windows-2016':
-      vcVars = '"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat"'
-      break
-    case 'windows-2019':
-      vcVars = '"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat"'
-      break
-    case 'windows-2022':
-      vcVars = '"C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat"'
-      break
-    default:
-      throw new Error(`Unknown Windows Image: ${virtualEnv}`)
+  let cmd = 'vswhere -latest -property installationPath'
+  let vcVars = `${cp.execSync(cmd).toString().trim()}\\VC\\Auxiliary\\Build\\vcvars64.bat`
+
+  if (!fs.existsSync(vcVars)) {
+    throw new Error(`Missing vcVars file: ${vcVars}`)
   }
   core.exportVariable('VCVARS', vcVars)
 
+  cmd = `cmd.exe /c ""${vcVars}" && set"`
```

Comment:
Are the double `"` intended here?

Reply:
Yes.  There are two pairs, an outer and an inner.  Kind of a pita when Actions uses PowerShell Core for the default script processor, but almost all languages (like Ruby & node.js ) shell out using cmd script.  The two have different styles for handling embedded quotes in commands...